src/main/kotlin/com/openlattice/graph/partioning/RepartitioningJob.kt

```diff
@@ -0,0 +1,315 @@
+package com.openlattice.graph.partioning
+
+import com.fasterxml.jackson.annotation.JsonCreator
+import com.geekbeast.rhizome.jobs.AbstractDistributedJob
+import com.hazelcast.core.HazelcastInstance
+import com.hazelcast.map.IMap
+import com.openlattice.edm.EntitySet
+import com.openlattice.hazelcast.HazelcastMap
+import com.openlattice.hazelcast.serializers.decorators.MetastoreAware
+import com.openlattice.postgres.DataTables.*
+import com.openlattice.postgres.PostgresArrays
+import com.openlattice.postgres.PostgresColumn.*
+import com.openlattice.postgres.PostgresColumnDefinition
+import com.openlattice.postgres.PostgresTable.*
+import com.openlattice.postgres.PostgresTableDefinition
+import com.openlattice.postgres.ResultSetAdapters
+import com.zaxxer.hikari.HikariDataSource
+import java.sql.PreparedStatement
+import java.util.*
+
+/**
+ * Background job for re-partitioning data in an entity set.
+ *
+ * @author Matthew Tamayo-Rios &lt;matthew@openlattice.com&gt;
+ */
+class RepartitioningJob
+@JsonCreator constructor(
+        state: RepartitioningJobState
+) : AbstractDistributedJob<Long, RepartitioningJobState>(state), MetastoreAware {
+    constructor(
+            entitySetId: UUID,
+            oldPartitions: List<Int>,
+            newPartitions: Set<Int>
+    ) : this(RepartitioningJobState(entitySetId, oldPartitions,newPartitions))
+    private var phase = 0
+
+    @Transient
+    private lateinit var hds: HikariDataSource
```

Comment:
Does Kotlin not have a runtime exception here b/c there's no init block?

Reply:
Why would there be a run time exception? On the client-side it's not an issue unless someone invokes call() without calling setHikariDataSource. Otherwise, the stream serializer initializes the value before it's ever used.

Reply:
Oh derp, I hadn't caught the full context for it. I was pretty sure that lateinit variables cause runtime exceptions without init blocks but it looks like it would be thrown at read time 😅